homeassistant/components/modbus/sensor.py

```diff
@@ -121,11 +122,12 @@ async def async_setup_platform(
             del entry[CONF_REGISTER_TYPE]
 
     for entry in discovery_info[CONF_SENSORS]:
+        total_reg_size = entry[CONF_COUNT] * entry.get(CONF_REGISTER_SIZE, 2)
         if entry[CONF_DATA_TYPE] == DATA_TYPE_STRING:
-            structure = str(entry[CONF_COUNT] * 2) + "s"
+            structure = str(total_reg_size) + "s"
         elif entry[CONF_DATA_TYPE] != DATA_TYPE_CUSTOM:
             try:
-                structure = f">{DEFAULT_STRUCT_FORMAT[entry[CONF_DATA_TYPE]][entry[CONF_COUNT]]}"
```

Comment:
What was wrong with DEFAULT_STRUCT_FORMAT (which still are in const.py) I do not understand the need for 2 identical (or near identical) structures.